pkg/storage/cacher.go

```diff
@@ -417,18 +401,13 @@ func filterFunction(key string, keyFunc func(runtime.Object) (string, error), fi
 
 // Returns resource version to which the underlying cache is synced.
 func (c *Cacher) LastSyncResourceVersion() (uint64, error) {
-	// To avoid situation when LastSyncResourceVersion is processed before the
-	// underlying watchCache is propagated, we acquire 'usable' lock.
-	c.usable.RLock()
-	defer c.usable.RUnlock()
-
-	c.RLock()
-	defer c.RUnlock()
+	c.ready.wait()
```

Comment:
I am not sure why exactly we need to hold the c.Rlock. It would be great if @wojtek-t can help to review. Also if it is really tricky, I would love to write a test to ensure we make the tricky part right.